src/client.ts

```diff
@@ -1,121 +1,197 @@
       });
     }
   }
 
-  /**
-   * Iterates each user resource in the provider.
-   *
-   * @param iteratee receives each resource to produce entities/relationships
-   */
-  public async iterateUsers(
-    iteratee: ResourceIteratee<AcmeUser>,
-  ): Promise<void> {
-    // TODO paginate an endpoint, invoke the iteratee with each record in the
-    // page
-    //
-    // The provider API will hopefully support pagination. Functions like this
-    // should maintain pagination state, and for each page, for each record in
-    // the page, invoke the `ResourceIteratee`. This will encourage a pattern
-    // where each resource is processed and dropped from memory.
-
-    const users: AcmeUser[] = [
-      {
-        id: 'acme-user-1',
-        name: 'User One',
-      },
-      {
-        id: 'acme-user-2',
-        name: 'User Two',
-      },
-    ];
-
-    for (const user of users) {
-      await iteratee(user);
-    }
+  public async getCurrentUser(): Promise<User> {
+    const endpoint = this.withBaseUri(
+      '/users/me?opt_fields=gid,name,email,resource_type',
+    );
+    const response = await this.request<Payload<User>>(endpoint);
+    return response.data;
   }
 
-  /**
-   * Iterates each group resource in the provider.
-   *
-   * @param iteratee receives each resource to produce entities/relationships
-   */
-  public async iterateGroups(
-    iteratee: ResourceIteratee<AcmeGroup>,
-  ): Promise<void> {
-    // TODO paginate an endpoint, invoke the iteratee with each record in the
-    // page
-    //
-    // The provider API will hopefully support pagination. Functions like this
-    // should maintain pagination state, and for each page, for each record in
-    // the page, invoke the `ResourceIteratee`. This will encourage a pattern
-    // where each resource is processed and dropped from memory.
-
-    const groups: AcmeGroup[] = [
-      {
-        id: 'acme-group-1',
-        name: 'Group One',
-        users: [
-          {
-            id: 'acme-user-1',
-          },
-        ],
-      },
-    ];
-
-    for (const group of groups) {
-      await iteratee(group);
-    }
+  public async iterateUsers(iteratee: ResourceIteratee<User>) {
+    const path = '/users?opt_fields=gid,name,email,resource_type';
+
+    const endpoint = this.withBaseUri(path);
+    const res = await this.request<Payload<User[]>>(endpoint);
+    const userList = res.data;
+
+    for (const user of userList) await iteratee(user);
+  }
+
+  public async iterateWorkspaces(iteratee: ResourceIteratee<Workspace>) {
+    let body: Paginated<Workspace[]>;
+    let path = `/workspaces?limit=${this.paginateEntitiesPerPage}&opt_fields=gid,name,resource_type`;
+    let endpoint: string;
+
+    do {
+      endpoint = this.withBaseUri(path);
+      body = await this.request<Paginated<Workspace[]>>(endpoint);
+      for (const workspace of body.data) await iteratee(workspace);
+      if (body.next_page !== null) path = body.next_page.path;
```

Comment:
We would like to see these changed to the following (throughout the code):

```suggestion
      for (const workspace of body.data) {
         await iteratee(workspace);
      }
      if (body.next_page !== null) {
         path = body.next_page.path;
      }
```

It's just all other integrations follow that style so we'd like to stick with that.

Reply:
Sure!